From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Revert "hwmon: (coretemp) Fix oops on driver load"
Patch-mainline: Never, SUSE-specific

This is a revert of the patch in the 3.0.13 release.

We have coretemp driver updates in our tree already that this ends up
conflicting with, so Jean said to revert this as it is safe for us to do
here.

Cc: Jean Delvare <jdelvare@suse.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

--- b/drivers/hwmon/coretemp.c
+++ a/drivers/hwmon/coretemp.c
@@ -539,8 +539,6 @@
 		return;
 
 	pdata = platform_get_drvdata(pdev);
-	if (!pdata)
-		return;
 
 	err = create_core_data(pdata, pdev, cpu, pkg_flag);
 	if (err)
